### ISSUE TITLE
Support multi-value counters

### ISSUE THREAD

**Issue by higels** [leoluk/perflib_exporter#10]
As in my other issue, go 1.11, perflib exporter from master I pulled down today, Server 2016, Dell R730 and FC630.

I'm interested in the data from objects 510 (Network Interface) and 1820 (Network Adapter). When I enable them individually like so:

```
perflib_exporter.exe --telemetry.addr=":9100" --log.level="debug" --perflib.objects=510
```
I do not get much in the way of outputted metrics (see below). Oddly, they both seem to produce exactly the same metrics.

I get some debug spew indicating that it is querying interfaces (attached).

I've also included the dump?query=blah files, since I guess those are useful.

Interestingly, I've disabled all of the inactive interfaces in devmgmt.msc to see if it helped at all, but they still seem to be polled.

I would try to investigate and fix this myself, but I'm in a bit of a time crunch at the moment, so can't devote too much time to it for the next month or so.

```
perflib_exporter_build_info{branch="",goversion="go1.11",revision="",version=""} 1
perflib_exporter_collector_duration_seconds{collector="perflib"} 0.0390378
perflib_exporter_collector_success{collector="perflib"} 1
perflib_icmp_received_address_mask{name=""} 0
perflib_icmp_received_address_mask_reply{name=""} 0
perflib_icmp_received_source_quench{name=""} 0
perflib_icmp_received_timestamp_reply_total{name=""} 0
perflib_icmp_received_timestamp_total{name=""} 0
perflib_icmp_sent_address_mask{name=""} 0
perflib_icmp_sent_address_mask_reply{name=""} 0
perflib_icmp_sent_source_quench{name=""} 0
perflib_icmp_sent_timestamp_reply_total{name=""} 0
perflib_icmp_sent_timestamp_total{name=""} 0
perflib_icmpv6_received_membership_query_total{name=""} 0
<snip a bunch of icmpv6 metrics>
```

[perflib-510-debug-output.txt](https://github.com/leoluk/perflib_exporter/files/2373022/perflib-510-debug-output.txt)

[perflib-exporter-510.html.txt](https://github.com/leoluk/perflib_exporter/files/2373037/perflib-exporter-510.html.txt)

[perflib-exporter-1820.html.txt](https://github.com/leoluk/perflib_exporter/files/2373038/perflib-exporter-1820.html.txt)

**Comment by andywiens** [comment]
All performance counters with more than one internal counter are ignored. I've noticed a few other objects (e.g. logical disk) have valuable metrics that are excluded for the same reason.

@leoluk do you have ideas for how to support multi-value counters? My team is in the process of replacing our WMI exporter installations with this exporter and can devote some development time to help out. Wanted to see if you already have a solution in progress before we dig into it.

**Comment by leoluk** [comment]
Pasting from an email exchange I had with Anthony:

> Do you have a specific example of a multi-counter metric where the second metric contains a useful value?
>
>Right now, we just ignore them altogether, but that's just because I haven't decided how to deal with them. We can either:
>
> 1. pick the first value and ignore the second
> 2. generate different metrics for each value
>
> or probably a combination of both, but that really depends on what these values are.

**Comment by bismarck** [comment]
I took a stab at adding multi-value counter support in #14.  I figured we didn't need to support all counter types (e.g. PERF_COUNTER_NODATA, PERF_AVERAGE_BULK), so my change explicitly filters them out.  To help me decide what counter types to support I built a frequency distribution for the Global query of counter types and got the following results:

**Windows Server 2016**

Counter Type | Frequency
--------------|-----------
PERF_COUNTER_RAWCOUNT (0x00010000) | 12883
PERF_100NSEC_TIMER (0x20510500) | 5952
PERF_COUNTER_NODATA (0x40000200) | 5610
PERF_COUNTER_COUNTER (0x10410400) | 4416
PERF_COUNTER_BULK_COUNT (0x10410500) | 3263
PERF_COUNTER_LARGE_RAWCOUNT (0x00010100) | 2065
PERF_ELAPSED_TIME (0x30240500) | 1866
PERF_COUNTER_RAWCOUNT_HEX (0x00000000) | 1726
PERF_AVERAGE_BASE (0x40030402) | 336
PERF_AVERAGE_BULK (0x40020500) | 260
PERF_PRECISION_TIMESTAMP (0x40030500) | 156
PERF_PRECISION_100NS_TIMER (0x20570500) | 126
PERF_SAMPLE_BASE (0x40030401) | 96
PERF_SAMPLE_FRACTION (0x20c20400) | 96
PERF_AVERAGE_TIMER (0x30020400) | 76
PERF_RAW_FRACTION (0x20020400) | 63
PERF_RAW_BASE (0x40030403) | 63
PERF_COUNTER_100NS_QUEUELEN_TYPE (0x00550500) | 60
PERF_100NSEC_TIMER_INV (0x21510500) | 53
PERF_LARGE_RAW_FRACTION (0x20020500) | 27
PERF_COUNTER_TIMER (0x20410500) | 20
PERF_COUNTER_DELTA (0x00400400) | 4
PERF_PRECISION_SYSTEM_TIMER (0x20470500) | 3

**Windows Server 2019**

Counter Type | Frequency
--------------|-----------
PERF_COUNTER_RAWCOUNT (0x00010000) | 16671
PERF_100NSEC_TIMER (0x20510500) | 7679
PERF_COUNTER_COUNTER (0x10410400) | 4917
PERF_COUNTER_BULK_COUNT (0x10410500) | 4508
PERF_ELAPSED_TIME (0x30240500) | 2443
PERF_COUNTER_RAWCOUNT_HEX (0x00000000) | 2296
PERF_COUNTER_NODATA (0x40000200) | 2236
PERF_COUNTER_LARGE_RAWCOUNT (0x00010100) | 2155
PERF_AVERAGE_BASE (0x40030402) | 261
PERF_AVERAGE_BULK (0x40020500) | 185
PERF_PRECISION_TIMESTAMP (0x40030500) | 144
PERF_PRECISION_100NS_TIMER (0x20570500) | 126
PERF_RAW_BASE (0x40030403) | 78
PERF_RAW_FRACTION (0x20020400) | 78
PERF_AVERAGE_TIMER (0x30020400) | 76
PERF_COUNTER_100NS_QUEUELEN_TYPE (0x00550500) | 61
PERF_100NSEC_TIMER_INV (0x21510500) | 53
PERF_COUNTER_TIMER (0x20410500) | 38
PERF_SAMPLE_FRACTION (0x20c20400) | 36
PERF_SAMPLE_BASE (0x40030401) | 36
PERF_LARGE_RAW_FRACTION (0x20020500) | 15
PERF_COUNTER_DELTA (0x00400400) | 4
PERF_PRECISION_SYSTEM_TIMER (0x20470500) | 3

Given the above data, we should probably support the following counters:
```
PERF_COUNTER_RAWCOUNT_HEX
PERF_COUNTER_LARGE_RAWCOUNT_HEX
PERF_COUNTER_RAWCOUNT
PERF_COUNTER_LARGE_RAWCOUNT
PERF_COUNTER_DELTA
PERF_COUNTER_COUNTER
PERF_COUNTER_BULK_COUNT
PERF_RAW_FRACTION
PERF_LARGE_RAW_FRACTION
PERF_100NSEC_TIMER
PERF_PRECISION_100NS_TIMER
PERF_SAMPLE_FRACTION
PERF_100NSEC_TIMER_INV
PERF_ELAPSED_TIME
PERF_SAMPLE_BASE
PERF_RAW_BASE
PERF_LARGE_RAW_BASE
```

In order to make metric names unique, `_max` is appended to base counter types.  I also added code to explicitly map a counter type to a prometheus metric type (the existing code exposed everything as a counter).

**Comment by leoluk** [comment]
Resolved by #14. Ty!
